datanode/networkhistory/store/store.go

```diff
@@ -181,6 +183,9 @@ func New(ctx context.Context, log *logging.Logger, chainID string, cfg Config, n
 		return nil, fmt.Errorf("failed to setup metrics:%w", err)
 	}
 
+	// this can error if there are no existing segments so don't check the result.
+	p.CollectGarbage(ctx)
```

Comment:
sure happy to add but do you think it might be a bit confusing to have something like 

```
failed to remove old history segments:failed to get latest segment:index entry not found
```

in your first run of datanode? I guess I should stop being lazy and actually just check, I'll do that

Reply:
lol, yea I guess it would look weird. now I'm thinking as well. seeing how we check for time, the first time you start the datanode, as soon as you create a segment it will run it anyway, so maybe no need for this initial call?

Reply:
yes that's true, it will do it on the first segment; but fryd mentioned it would be nice to do it at startup.. I've altered it slightly now so that it won't error in the simple case that there are no segments.. we can keep that or just get rid of the initial call. 

this gives a slighly new behavior in that if there are some segments, but garbage collection fails in some other way, datanode won't start... which I can't quite work out if it's a bug or a feature